### PR TITLE
Feature/peer dependency

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* #944

*Description of changes:* CDK dependency is set as `peerDependencies` to allow for smoother installation and upgrade process. Documentation (Getting started) updated for common troubleshooting cases when installing and upgrading. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by elamaran11 — DISMISSED** [review]
LGTM. Thanks for the peer dependency documentation.

File: lib/addons/karpenter/index.ts [hunk 1]
```diff
@@ -469,6 +469,10 @@ export class KarpenterAddOn extends HelmAddOn {
 
         karpenterChart.node.addDependency(ns);
 
+        if(clusterInfo.nodeGroups) {
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:472]: @youngjeong46 this is what I added to fix. I was able to run it once e2e in the target account, I will run one more time. 
This code should only refence the node group (or fargate profile) used to deploy karpenter. I added deps on all node groups.

**Pull request merged**